src/systems/systems/src/hydraulic/trimmable_horizontal_stabilizer.rs

```diff
@@ -0,0 +1,1066 @@
+            && self.speed.get::<radian_per_second>() < 0.
+            || ths_hydraulic_assembly.is_at_max_up_spool_valve
+                && self.speed.get::<radian_per_second>() > 0.
+        {
+            self.speed = AngularVelocity::default();
+        }
+    }
+
+    fn elec_motor_drive_total_speed(&self) -> AngularVelocity {
+        let mut sum_of_speeds = AngularVelocity::default();
+
+        for (motor_index, motor) in self.electric_motors.iter().enumerate() {
+            if self.electric_clutches[motor_index].is_clutch_engaged() {
+                sum_of_speeds +=
+                    motor.speed() / self.elec_motor_over_trim_actuator_ratio.get::<ratio>();
+            }
+        }
+
+        sum_of_speeds
+    }
+
+    fn update_clutches_state(&mut self, controller: &impl PitchTrimActuatorController) {
+        for (clutch_index, clutch) in self.electric_clutches.iter_mut().enumerate() {
+            clutch.set_is_powered(self.electric_motors[clutch_index].is_powered());
+            clutch.set_energized(controller.energised_motor()[clutch_index]);
+        }
+    }
+
+    fn update_motors(
+        &mut self,
+        context: &UpdateContext,
+        controller: &impl PitchTrimActuatorController,
+        ths_hydraulic_assembly: &TrimmableHorizontalStabilizerHydraulics,
+    ) {
+        for (motor_index, motor) in self.electric_motors.iter_mut().enumerate() {
+            motor.set_active_state(controller.energised_motor()[motor_index]);
+
+            let trim_actuator_normalized_position_request = ths_hydraulic_assembly
+                .normalized_position_from_ths_deflection(controller.commanded_position());
+
+            let final_trim_actuator_position_request = trim_actuator_normalized_position_request
+                .get::<ratio>()
+                * (self.max_actuator_angle - self.min_actuator_angle)
+                + self.min_actuator_angle;
+
+            motor.update(context, self.position, final_trim_actuator_position_request);
+        }
+    }
+
+    fn position_normalized(&self) -> Ratio {
+        let range = self.max_actuator_angle - self.min_actuator_angle;
+
+        Ratio::new::<ratio>(
+            (self.position.get::<radian>() - self.min_actuator_angle.get::<radian>())
+                / range.get::<radian>(),
+        )
+        .min(Ratio::new::<ratio>(100.))
+        .max(Ratio::new::<ratio>(0.))
+    }
+}
+impl SimulationElement for PitchTrimActuator {
+    fn accept<T: SimulationElementVisitor>(&mut self, visitor: &mut T) {
+        accept_iterable!(self.electric_motors, visitor);
+        visitor.visit(self);
+    }
+
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(&self.manual_override_id, self.manual_override_active);
+    }
+}
+
+pub struct TrimmableHorizontalStabilizerAssembly {
+    pitch_trim_actuator: PitchTrimActuator,
+    trim_wheel: TrimWheels,
+    ths_assembly: TrimmableHorizontalStabilizerHydraulics,
```

Comment:
I don't think this should be called `ths_assembly` as it is not actually of type `TrimmableHorizontalStabilizerAssembly`, which is what I thought of when I saw `ths_assembly`.